From 9ab6f8fd6af5fe099246becbd9ddcf527e76fedb Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Sun, 30 Mar 2025 17:47:05 +0200
Subject: [PATCH 8/8] feat: Add Patch SMSC setting to Treble App

Requires a personal patch in platform_frameworks_opt_telephony.
Primarily intended for Samsung devices. Some require turning this off.

Change-Id: I5595648430907090e7a6784a56898a60fd21998f
---
 app/src/main/java/me/phh/treble/app/Misc.kt         | 4 ++++
 app/src/main/java/me/phh/treble/app/MiscSettings.kt | 1 +
 app/src/main/res/xml/pref_misc.xml                  | 5 +++++
 3 files changed, 10 insertions(+)

diff --git a/app/src/main/java/me/phh/treble/app/Misc.kt b/app/src/main/java/me/phh/treble/app/Misc.kt
index 7c0c1c9..9183f32 100644
--- a/app/src/main/java/me/phh/treble/app/Misc.kt
+++ b/app/src/main/java/me/phh/treble/app/Misc.kt
@@ -242,6 +242,10 @@ object Misc: EntryStartup {
                 val value = sp.getBoolean(key, false)
                 SystemProperties.set("persist.sys.phh.restart_ril", if (value) "true" else "false")
             }
+            MiscSettings.patchSmsc -> {
+                val value = sp.getBoolean(key, true)
+                SystemProperties.set("persist.sys.phh.patch_smsc", if (value) "true" else "false")
+            }
             MiscSettings.minimalBrightness -> {
                 val value = sp.getBoolean(key, false)
                 SystemProperties.set("persist.sys.overlay.minimal_brightness", if (value) "true" else "false")
diff --git a/app/src/main/java/me/phh/treble/app/MiscSettings.kt b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
index e80012a..2a434ea 100644
--- a/app/src/main/java/me/phh/treble/app/MiscSettings.kt
+++ b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
@@ -34,6 +34,7 @@ object MiscSettings : Settings {
     val backlightScale = "key_misc_backlight_scale"
     val headsetDevinput = "key_misc_headset_devinput"
     val restartRil = "key_misc_restart_ril"
+    var patchSmsc = "key_misc_patch_smsc"
     val minimalBrightness = "key_misc_minimal_brightness"
     val aod = "key_misc_aod"
     val dt2w = "key_misc_dt2w"
diff --git a/app/src/main/res/xml/pref_misc.xml b/app/src/main/res/xml/pref_misc.xml
index a8221a7..4d56605 100644
--- a/app/src/main/res/xml/pref_misc.xml
+++ b/app/src/main/res/xml/pref_misc.xml
@@ -179,6 +179,11 @@
             android:defaultValue="false"
             android:key="key_misc_restart_ril"
             android:title="Automatically restart RIL" />
+        <SwitchPreference
+            android:defaultValue="true"
+            android:key="key_misc_patch_smsc"
+            android:title="Patch SMSC"
+            android:summary="If you can't send messages, primarily on Samsung devices, turn off SMSC patching" />
         <Preference android:title="@string/remove_telephony_subsystem"
             android:key="key_misc_removetelephony" />
     </PreferenceCategory>
--
